src/App.tsx

```diff
@@ -8,37 +8,24 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { setSelectedPost } from './features/selectedPost/selectedPostSlice';
+import { loadPosts, setPosts } from './features/posts/postsSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const { items, loaded, hasError } = useAppSelector(store => store.posts);
+  const { selectedPost } = useAppSelector(store => store.selectedPost);
+  const { author } = useAppSelector(store => store.author);
 
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
-
-  function loadUserPosts(userId: number) {
-    setLoaded(false);
-
-    getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
-  }
+  const dispatch = useAppDispatch();
 
   useEffect(() => {
-    // we clear the post when an author is changed
-    // not to confuse the user
-    setSelectedPost(null);
+    dispatch(setSelectedPost(null));
 
     if (author) {
-      loadUserPosts(author.id);
+      dispatch(loadPosts(author.id));
     } else {
-      setPosts([]);
+      dispatch(setPosts([]));
     }
   }, [author]);
```

Comment:
do not ignore lint suggestions - it's better to do it:
![image](https://github.com/mate-academy/react_redux-list-of-posts/assets/86923089/505c52af-d25f-42b2-94ec-73be3bee13bc)